sdk/src/main/java/ch/admin/bag/covidcertificate/sdk/android/verification/task/VerifierCertificateVerificationTask.kt

```diff
@@ -0,0 +1,64 @@
+ *
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+package ch.admin.bag.covidcertificate.sdk.android.verification.task
+
+import android.net.ConnectivityManager
+import ch.admin.bag.covidcertificate.sdk.android.models.VerifierCertificateHolder
+import ch.admin.bag.covidcertificate.sdk.core.models.state.CheckNationalRulesState
+import ch.admin.bag.covidcertificate.sdk.core.models.state.VerificationState
+import ch.admin.bag.covidcertificate.sdk.core.models.trustlist.TrustList
+import ch.admin.bag.covidcertificate.sdk.core.verifier.CertificateVerifier
+
+/**
+ * The verification task implementation specific for verifier applications. This task takes a [VerifierCertificateHolder] and
+ * returns a reduced verification state that does not contain information that could indicate what type of certificate was verified.
+ *
+ * @param certificateHolder The verifier certificate holder (containing the certificate data) to verify
+ * @param connectivityManager The Android connectivity service used to check if the device is offline or not
+ * @param ignoreLocalTrustList True to ignore the local trust list during verification and force either an offline or network error
+ */
+internal class VerifierCertificateVerificationTask(
+	private val certificateHolder: VerifierCertificateHolder,
+	connectivityManager: ConnectivityManager,
+	ignoreLocalTrustList: Boolean = false
+) : CertificateVerificationTask(connectivityManager, ignoreLocalTrustList) {
+
+	override suspend fun verify(verifier: CertificateVerifier, trustList: TrustList): VerificationState {
+		val verificationState = verifier.verify(certificateHolder.certificateHolder, trustList)
+		return when (verificationState) {
+			is VerificationState.LOADING -> verificationState
+			is VerificationState.SUCCESS -> verificationState
+			is VerificationState.INVALID -> {
+				// Strip out all information that identifies the national rule that failed for verifier apps
+				when (val nationalRulesState = verificationState.nationalRulesState) {
+					is CheckNationalRulesState.SUCCESS -> {
+						val strippedNationalRulesState = nationalRulesState.copy(validityRange = null)
+						verificationState.copy(nationalRulesState = strippedNationalRulesState)
+					}
+					is CheckNationalRulesState.NOT_YET_VALID -> {
+						val strippedNationalRulesState = nationalRulesState.copy(validityRange = null, ruleId = null)
+						verificationState.copy(nationalRulesState = strippedNationalRulesState)
+					}
+					is CheckNationalRulesState.NOT_VALID_ANYMORE -> {
+						val strippedNationalRulesState = nationalRulesState.copy(validityRange = null, ruleId = null)
+						verificationState.copy(nationalRulesState = strippedNationalRulesState)
+					}
+					is CheckNationalRulesState.INVALID -> {
+						val strippedNationalRulesState = nationalRulesState.copy(nationalRulesError = null, ruleId = null)
+						verificationState.copy(nationalRulesState = strippedNationalRulesState)
+					}
```

Comment:
Couldn't this be shortened by unifying the repeated copying into one?
```kotlin
is NOT_YET_VALID, is NOT_VALID_ANYMORE, is INVALID -> {
// copy
}
```

Reply:
No, because afterwards I access the properties of the individual states. If I unify the repeated when clauses, the smart casting wouldn't work anymore